appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/StatelessSessionContainer.java

```diff
@@ -248,17 +252,26 @@ protected void initializeHome()
     protected void createBeanPool() {
         ObjectFactory sessionCtxFactory = new SessionContextFactory();
 
-                iased = ejbDescriptor.getIASEjbExtraDescriptors();
+        iased = ejbDescriptor.getIASEjbExtraDescriptors();
         if( iased != null) {
             beanPoolDes = iased.getBeanPool();
         }
 
         poolProp = new PoolProperties(ejbContainer, beanPoolDes);
         String val = ejbDescriptor.getEjbBundleDescriptor().getEnterpriseBeansProperty(SINGLETON_BEAN_POOL_PROP);
-        pool= new NonBlockingPool(getContainerId(), ejbDescriptor.getName(),
-           sessionCtxFactory, poolProp.steadyPoolSize,
-           poolProp.poolResizeQuantity, poolProp.maxPoolSize,
-           poolProp.poolIdleTimeoutInSeconds, loader, Boolean.parseBoolean(val));
+        if(poolProp.maxWaitTimeInMillis != -1) {
```

Comment:
Yes. Default is the unchanged behavior. I tried to document it in the github issue and in the Payara issue